File: tests/csharp/tests-csharp-xunit-api_request-playwright/Helpers/DataFaker.cs
namespace Helpers;

/// <summary>Throwaway test identities. Username fits backend Size(min = 3, max = 64).</summary>
public static class DataFaker
{
    public static string Username() => $"user_{Guid.NewGuid():N}"[..16];

    public static string Password() => "password123";

    /// <summary>Exactly Size(min = 3) — unique hex slice.</summary>
    public static string UsernameAtMinLength() => Guid.NewGuid().ToString("N")[..3];

    /// <summary>Exactly Size(min = 6).</summary>
    public static string PasswordAtMinLength() => "123456";
}

File: tests/java/tests-java-junit5-rest_assured-selenide-appium/src/test/java/helpers/DataFaker.java
package helpers;

import java.util.UUID;

/**
 * Throwaway identities for native e2e. The web cell keeps DataFaker/UserBuilder;
 * this is the slice Login / Register / DeleteAccount need.
 */
public final class DataFaker {

    private DataFaker() {
    }

    /** Exactly backend {@code @Size(min = 3)} — unique hex slice. */
    public static String usernameAtMinLength() {
        return UUID.randomUUID().toString().replace("-", "").substring(0, 3);
    }

    /** Exactly backend {@code @Size(min = 6)}. */
    public static String passwordAtMinLength() {
        return "123456";
    }

    /** Unique login within backend {@code @Size(min = 3, max = 64)}. */
    public static String username() {
        return "u" + UUID.randomUUID().toString().replace("-", "").substring(0, 12);
    }

    public static String password() {
        return "password123";
    }
}
